homeassistant/components/ads/__init__.py

```diff
@@ -78,7 +88,7 @@
 )
 
 
-def setup(hass, config):
+def setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
The other possibility is to add `# type: ignore[attr-defined]` everywhere, but it seems a shame not to move the constants out of `AdsHub`.